### DISCUSSION TITLE
Update to docs to show what's available

### DISCUSSION THREAD

**Discussion by kurucu** [filamentphp/filament#3178] [Ideas]
Hi Team,

I created a reactive callback to update the slug only when the record is new. Didn't know how to do it, recalled seeing something, gave this a go and it worked:

```php
->afterStateUpdated(function (\Closure $set, $state, \Closure $get) {
    if (!$get('id')) {
        $set('slug', Str::slug($state));
    }
}),
```

The thing is I can only tell by guessing, whilst looking at Filament's code, why it works. It would be GREAT if the docs covered stuff like this please? If I knew what was going on, I'd submit a PR 😂

**Comment by danharrin** [comment]
Personally I wouldnt do it like that, instead pass in the $livewire property and check if $livewire is instanceof CreateRecord instead?

There are way too many code examples like this that will not fit into the documentation. Thats why we have a Tricks section on our website! You can log in and submit this code example with a short explanation to share it with others.

**Reply by kurucu** [reply]
Thanks for the advice, I'll take that.

Also forgot about the tricks section - will get it working and submit!
